roles/tests-integration/lib/sniffer.rs

```diff
@@ -59,7 +59,47 @@ pub struct Sniffer {
     messages_from_downstream: MessagesAggregator,
     messages_from_upstream: MessagesAggregator,
     check_on_drop: bool,
-    intercept_messages: Vec<InterceptMessage>,
+    actions: Vec<Action>,
```

Comment:
i would try to avoid the ordering stuff. we can do that by either using `action: Action` here or use a data type that provide a grantee over ordering

Reply:
the `Vec<Action>` is a nice to have but we never needed it really